modules/tensor_mechanics/src/materials/cohesive_zone_model/SlipWeakeningFriction3d.C

```diff
@@ -0,0 +1,182 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+/*
+Material Description of Slip Weakening Friction 3d
```

Comment:
As @lynnmunday mentioned, there are a lot of duplicated codes between 2d and 3d. Consider either have a common base class for 2d and 3d OR write a combined version which switches between 2d and 3d based the mesh dimensions.

Reply:
@chunhuizhao478 I think the common base class is the way to go